src/rust/engine/src/interning.rs

```diff
@@ -84,7 +82,13 @@ impl Interns {
       .read()
       .get(&k)
       .cloned()
-      .unwrap_or_else(|| panic!("Previously memoized object disappeared for {:?}", k))
+      .unwrap_or_else(|| {
+        panic!(
+          "Previously memoized object disappeared for Key {{ id: {}, type_id: {} }}!",
+          k.id(),
+          k.type_id()
+        )
```

Comment:
I was too late for this, but perhaps a comment explaining how we must not use `:?`?

Reply:
Done in #12167.